In [1]:
import pandas as pd
import numpy as np
from sklearn import linear_model

In [9]:
def multiple_regression(data):
    x_data = data[['Unsafe water source','Unsafe sanitation']]
    y_data = data['No access to handwashing facility']
    reg_obj = linear_model.LinearRegression()
    reg_obj.fit(x_data,y_data)
    literacy_predict = reg_obj.predict([[1000,3000]])
    return literacy_predict

def cov_mr(data):
    x_data = data[['Unsafe water source','Unsafe sanitation']]
    y_data = data['No access to handwashing facility']
    reg_obj = linear_model.LinearRegression()
    reg_obj.fit(x_data,y_data)
    return reg_obj.coef_

In [14]:
data = pd.read_csv('data_2.csv')
ind_data = data[data['Code']=='IND']
pak_data = data[data['Code']=='PAK']
print("India -----",cov_mr(ind_data))
print("Pakistan --",cov_mr(pak_data))

India ----- [0.53941374 0.12691466]
Pakistan -- [0.14601608 0.38339417]


From the regression coefficient it is clearly evident that India's water sources are contributing much to the countries handwashing facility than in Pakistan i.e., one unit downfall in the water source purity leads to 0.53 units decrease in access to handwash. In Pakistan, the sanitation is contributing much to the handwashing facility than in India i.e., one unit downfall in the sanitation facility leads to 0.38 units decrease in access to handwash.

The comparisions within the countries provides the clear insight that in India the water sources affect on handwashing facility is much more than the affect of sanitation on handwashing facility. On the other hand in Pakistan, the handwashing facility is degrading due to unhygienic sanitation.

Finally, it can be inferred that in order to increase the handwashing facility, the water sources need to be improved in India and sanitation need to be improved in Pakistan.